### PR TITLE
Create navigation

### PR DESCRIPTION
**Pull request by felire**
### Summary

Add react-navigation with 'withNavigation' to wrap the component Book into a component who has the navigation prop. Also, I add the TouchableOpacity Button into the Book component and add the BookDetail component empty. 
Add ROUTES constant and the navigation generic Proptype.

### Screenshots

transition2

### CODE REVIEW

**Comment by santinucera** [comment]
Remember to fix you `.gitignore` in order to not upload unnecessary files

File: WBooks/ios/WBooks.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -40,6 +39,8 @@
 		ADBDB9381DFEBF1600ED6528 /* libRCTBlob.a in Frameworks */ = {isa = PBXBuildFile; fileRef = ADBDB9271DFEBF0700ED6528 /* libRCTBlob.a */; };
 		ED297163215061F000B7C4FE /* JavaScriptCore.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = ED297162215061F000B7C4FE /* JavaScriptCore.framework */; };
 		ED2971652150620600B7C4FE /* JavaScriptCore.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = ED2971642150620600B7C4FE /* JavaScriptCore.framework */; };
+		6823AF0B11E14638B26FA724 /* libRNGestureHandler.a in Frameworks */ = {isa = PBXBuildFile; fileRef = A07E6A5D4DC2428BA899F2AB /* libRNGestureHandler.a */; };
```
Comment (santinucera) [inline WBooks/ios/WBooks.xcodeproj/project.pbxproj]: Remove all files from the repository

**Comment by santinucera** [comment]
Taking apart of the comments of the `.gitignore`, this a nice PR. Great work.

File: WBooks/ios/WBooks.xcodeproj/project.pbxproj (hunk 1 above)
Comment (felire) [reply WBooks/ios/WBooks.xcodeproj/project.pbxproj]: what files specifically? There are a lot of folders and i don't know which can i delete and which don't

Comment (felire) [reply WBooks/ios/WBooks.xcodeproj/project.pbxproj]: The vscode folder i suppose but in the folder android or ios i don't know.

**Comment by santinucera** [comment]
Approved ☑️

**Pull request merged**